source/common/ur_util.hpp

```diff
@@ -13,6 +13,28 @@
 #include <string.h>
 #include <string>
 
+/* for compatibility with non-clang compilers */
+#if defined(__has_feature)
+#define CLANG_HAS_FEATURE(x) __has_feature(x)
+#else
+#define CLANG_HAS_FEATURE(x) 0
+#endif
+
+/* define for running with address sanitizer */
+#if CLANG_HAS_FEATURE(address_sanitizer) || defined(__SANITIZE_ADDRESS__)
+#define SANITIZER_ADDRESS
+#endif
+
+/* define for running with memory sanitizer */
+#if CLANG_HAS_FEATURE(memory_sanitizer)
+#define SANITIZER_MEMORY
+#endif
+
+/* define for running with any sanitizer runtime */
+#if defined(SANITIZER_MEMORY) || defined(SANITIZER_ADDRESS)
```

Comment:
what about UBSAN?

Reply:
It does not require that we avoid `RTLD_DEEPBIND`. GCC also doesn't have `__SANITIZE_ADDRESS__`, and clang only recently added `__has_feature(undefined_behavior_sanitizer)`, so there's no reliable way of checking this anyway.

Reply:
👍